src/tests/BatchRendererTest.cpp

```diff
@@ -0,0 +1,699 @@
+      esp::gfx::BatchRendererStandaloneConfiguration{}
+          /* No QuietLog, to verify at least once that the log *is* printed */
+  };
+  // clang-format on
+
+  CORRADE_COMPARE(renderer.tileSize(), (Mn::Vector2i{48, 32}));
+  CORRADE_COMPARE(renderer.tileCount(), (Mn::Vector2i{2, 3}));
+  CORRADE_COMPARE(renderer.sceneCount(), 6);
+
+  for (std::size_t i = 0; i != renderer.sceneCount(); ++i) {
+    CORRADE_ITERATION(i);
+
+    /* Implicit camera transform, no object transforms in any scene */
+    CORRADE_COMPARE(renderer.camera(i), Mn::Matrix4{});
+    CORRADE_COMPARE(renderer.transformations(i).size(), 0);
+  }
+
+  /* Add a file, because that's currently required */
+  // TODO make it non-required (instantiate some empty shader if nothing)
+  renderer.addFile(Cr::Utility::Path::join(TEST_ASSETS, "scenes/batch.gltf"));
+
+  /* Nothing should be drawn, just the clear color */
+  renderer.draw();
+  Mn::Image2D color = renderer.colorImage();
+  Mn::Image2D depth = renderer.depthImage();
+  MAGNUM_VERIFY_NO_GL_ERROR();
+
+  /* Verify the color actually matches expectation so we don't need to manually
+     color pick the image every time it changes */
+  CORRADE_COMPARE_AS(
+      color,
+      Cr::Utility::Path::join(TEST_ASSETS,
+                              "screenshots/BatchRendererTestDefaults.png"),
+      Mn::DebugTools::CompareImageToFile);
+  CORRADE_COMPARE(color.size(), Mn::Vector2i{96});
+  CORRADE_COMPARE(color.format(), Mn::PixelFormat::RGBA8Unorm);
+  CORRADE_COMPARE(color.pixels<Mn::Color4ub>()[48][48], 0x1f1f1f_rgb);
+
+  /* Verify the depth readout works as well. Also should be a single value. */
+  CORRADE_COMPARE(depth.size(), Mn::Vector2i{96});
+  CORRADE_COMPARE(depth.format(), Mn::PixelFormat::Depth32F);
+  CORRADE_COMPARE(depth.pixels<Mn::Float>()[48][48], 1.0f);
+}
+
+void BatchRendererTest::singleMesh() {
+  // clang-format off
+  esp::gfx::BatchRendererStandalone renderer{
+      esp::gfx::BatchRendererConfiguration{}
+          .setTileSizeCount({128, 96}, {1, 1}),
+      esp::gfx::BatchRendererStandaloneConfiguration{}
+          .setFlags(esp::gfx::BatchRendererStandaloneFlag::QuietLog)
+  };
+  // clang-format on
+
+  renderer.addFile(Cr::Utility::Path::join(TEST_ASSETS, "scenes/batch.gltf"));
+
+  /* Undo the aspect ratio, move camera back */
+  renderer.camera(0) =
+      Mn::Matrix4::orthographicProjection(2.0f * Mn::Vector2{4.0f / 3.0f, 1.0f},
+                                          0.1f, 10.0f) *
+      Mn::Matrix4::translation(Mn::Vector3::zAxis(1.0f)).inverted();
+
+  CORRADE_COMPARE(renderer.add(0, "square"), 0);
+  /* It adds one transformation for the top-level object and then one nested
+     for the mesh, corresponding to the layout inside the glTF file */
+  CORRADE_COMPARE(renderer.transformations(0).size(), 2);
+  renderer.transformations(0)[0] = Mn::Matrix4::scaling(Mn::Vector3{0.8f});
+
+  renderer.draw();
+  Mn::Image2D color = renderer.colorImage();
+  MAGNUM_VERIFY_NO_GL_ERROR();
```

Comment:
No :) Error checking is expensive, potentially causing driver stalls, and if the code causes GL errors, it's broken.

But I customarily do this in all test code because it helps me catch errors easier. In this case this check uncovered a NVidia-specific bug causing wrong camera matrix to be used, which I subsequently fixed in faad5b2fa195d7332c34447b7cdc3e781128546c.